bears/general/LineCountBear.py

```diff
@@ -17,16 +17,37 @@ def _get_blank_line_count(self, file):
             list(filter(lambda x: re.match(r'^\s*$', x), file)))
         return num_blank_lines
 
+    def _check_correct_range(self, min_lines_per_file, max_lines_per_file):
+        if min_lines_per_file > max_lines_per_file:
+            print(min_lines_per_file, max_lines_per_file)
+            raise ValueError('Allowed maximum lines per file({}) is smaller '
+                             'than minimum lines per file({}) and this is '
+                             'prohibited'.format(max_lines_per_file,
+                                                 min_lines_per_file))
+        return
+
     def run(self, filename, file, max_lines_per_file: int,
-            exclude_blank_lines: bool=False):
+            exclude_blank_lines: bool=False,
+            min_lines_per_file: int=0):
         """
-        Count the number of lines in a file and ensure that they are
-        smaller than a given size.
+        Count the number of lines in a file and ensure that they lie within
+        the range of given sizes.
 
-        :param max_lines_per_file: Number of lines allowed per file.
-        :param exclude_blank_lines: "True" if blank lines are to be excluded.
+        :param max_lines_per_file: Maximum number of lines allowed per file.
+        :param exclude_blank_lines: ``True`` if blank lines are to be excluded.
+        :param min_lines_per_file: Minimum number of lines required per file.
         """
         file_length = len(file)
+        try:
+            self._check_correct_range(min_lines_per_file, max_lines_per_file)
+        except ValueError as e:
+            yield Result.from_values(
+                origin=self,
+                message=str(e),
+                severity=RESULT_SEVERITY.NORMAL,
+                file=filename)
+            return
```

Comment:
No that's not a result, let the exception bubble up.

(We maybe have to implement somewhen a special exception for such errors, so the user gets a nice message instead of a traceback... Though probably those checks will then happen inside `__init__` then... Not sure yet)

Reply:
@Makman2 I have a small problem. I made something like this : 

`
if min_lines_per_file > max_lines_per_file:  
 raise ValueError('Allowed maximum lines per file({}) is smaller than minimum lines per file({}) and this is prohibited'.format(max_lines_per_file, min_lines_per_file)) 
`

And I made similar test to DartLintBear with use of assertRaisesRegex. However I had error.
How can I make test for this? :)

Test look like this : 

`def test_max_limit_is_smaller_than_min_limit(self):`

        EXCEPTION_MESSAGE = ('ValueError: Allowed maximum lines per '
                                'file(1) is smaller than minimum lines per '
                                'file(2) and this is prohibited')
        self.section.append(Setting('max_lines_per_file', 1))
        self.section.append(Setting('min_lines_per_file', 2))
        self.section.append(Setting('exclude_blank_lines', True))
        with self.assertRaisesRegex(AssertionError, EXCEPTION_MESSAGE):
            self.check_validity(self.uut, [])`

Reply:
Tbh I just read a different bear already using `logging.err` + `return`. I've thought about this again, and in fact the logging approach works too, even with caching. So take what you like, I'm gonna think about a uniform way to handle this later...